### ISSUE TITLE
examples do not build due to static asserts

### ISSUE THREAD

**Issue by Xeverous** [cycfi/elements#42]
In 3308ea63 I broke at least one example with new asserts.

```
In file included from [...]elements/element.hpp:11,
                 from [...]elements.hpp:12,
                 from [...]examples/buttons/main.cpp:6:
[...]elements/element/button.hpp: In instantiation of 'class cycfi::elements::basic_toggle_button<cycfi::elements::proxy<cycfi::elements::sprite&, cycfi::elements::basic_button> >':
[...]examples/buttons/main.cpp:101:64:   required from here
[...]elements/element/button.hpp:104:21: error: static assertion failed: basic_toggle_button Base type needs to be or inherit from layered_button
       static_assert(std::is_base_of_v<layered_button, Base>,
                     ^~~
```

Now I'm unsure what are actual template requirements. Apparently the default parameter is not the minimal requirement and the assert could be more permissive.

**Comment by djowel** [comment]
We should start integrating CI to prevent this from happening

**Comment by djowel** [comment]
I am reverting some of the astatic_asserts that are erroneous. For example, basic_toggle_button Base type needs to be or inherit from layered_button as long as it follows the required API and behavior (essentially generic programming).

**Comment by djowel** [comment]
Fixed in ebac2b21323c4643f1af4085e7ebe011ec9a5cc7
